lib/create-contentful-api.ts

```diff
@@ -751,43 +565,59 @@ export default function createContentfulApi<OptionType extends ChainOptions>(
   async function makeGetEntries<Fields extends FieldsType>(
     query,
     options: ChainOptions = {
-      withoutLinkResolution: false,
       withAllLocales: false,
+      withoutLinkResolution: false,
+      withoutUnresolvableLinks: false,
     }
   ) {
     const { withAllLocales } = options
 
-    validateLocaleParam(query, withAllLocales as boolean)
+    validateLocaleParam(query, withAllLocales)
     validateResolveLinksParam(query)
 
-    if (isClientWithAllLocalesAndWithoutLinkResolution(options)) {
-      return getEntriesWithAllLocalesAndWithoutLinkResolution<Fields>(query)
-    }
-    if (isClientWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks(options)) {
-      return getEntriesWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks<Fields>(query)
-    }
-    if (isClientWithAllLocalesAndWithLinkResolutionAndWithoutUnresolvableLinks(options)) {
-      return getEntriesWithAllLocalesAndWithLinkResolutionAndWithoutUnresolvableLinks<Fields>(query)
-    }
-    if (isClientWithoutLinkResolution(options)) {
-      return getEntriesWithoutLinkResolution<Fields>(query)
-    }
-    if (isClientWithLinkResolutionAndWithUnresolvableLinks(options)) {
-      return getEntriesWithLinkResolutionAndWithUnresolvableLinks<Fields>(query)
-    }
-    if (isClientWithLinkResolutionAndWithoutUnresolvableLinks(options)) {
-      return getEntriesWithLinkResolutionAndWithoutUnresolvableLinks<Fields>(query)
-    }
-    return getEntriesDefault<Fields>(query)
+    return internalGetEntries<Fields, any, Extract<ChainOptions, typeof options>>(
+      withAllLocales
+        ? {
+            ...query,
+            locale: '*',
+          }
+        : query,
+      options
+    )
   }
 
-  async function internalGetEntries<RType>(
+  type ConfiguredEntry<
```

Comment:
Could those type definitions (and actually all types / interfaces from other non-type files) make their way into a separate types file / folder? I know they are local, but their size in their current shape is overwhelming to have in the main file.